Hive.NET.Persistence/FileHiveStorageProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Text.Json;
+using Hive.NET.Core.Api;
+using Hive.NET.Core.Configuration.Storage;
+using Hive.NET.Core.Extensions;
+
+namespace Hive.NET.Persistence;
+
+  public class FileHiveStorageProvider : IHiveStorageProvider
+    {
+        private readonly string _filePath;
+
+        public FileHiveStorageProvider(string filePath)
+        {
+            _filePath = filePath;
+            EnsureFileCreated();
+        }
+
+        public void UpsertHive(Guid id, Core.Components.Hive? hive)
```

Comment:
What is use case of calling UpsertHive(id, null)? If there is such scenario some  null checking could be added